mpa/modules/hooks/checkpoint_hook.py

```diff
@@ -51,13 +51,17 @@ def __init__(self,
     def after_train_epoch(self, runner):
         if not self.by_epoch or not self.every_n_epochs(runner, self.interval):
             return
-
         if hasattr(runner, 'save_ckpt'):
             if runner.save_ckpt:
+                if runner.save_ema_model:
+                    backup_model = runner.model
+                    runner.model = runner.ema_model
                 runner.logger.info(f'Saving checkpoint at {runner.epoch + 1} epochs')
                 if self.sync_buffer:
                     allreduce_params(runner.model.buffers())
                 self._save_checkpoint(runner)
+                if runner.save_ema_model:
+                    runner.model = backup_model
```

Comment:
Is EMA used in other task? If not - it is not the issue. 
If we want to enable EMA in other task - we should add this functionality in other way. We need to modify saving checkpoints function for each task. 
For now it works only with classification and doesn't affect other task since it won't have "ema_model" attribute

I propose to leave it as is for now. Because I can't see where we can simply put model substitution for every task.

Reply:
We are not using EMA hook for another task but I agree with you, let's leave it as is.